python/lsst/daf/butler/script/_pruneDatasets.py

```diff
@@ -18,14 +18,18 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
-
+from __future__ import annotations
 
 from enum import Enum, auto
+from typing import TYPE_CHECKING, Any, Callable, Iterable
```

Comment:
Same here, Callable and Iterable should come from `collections.abc`. Check other files too, I won't comment on them.